### PR TITLE
Webc 633 Shopify Admin Api client integration

### PR DESCRIPTION
**Pull request by sofib**

### CODE REVIEW

File: extension/lib/shopgate/CategoryExtensionPipeline.js [hunk 1]
```diff
@@ -30,8 +30,8 @@ class ShopgateCategoryExtensionPipeline {
    * @returns {ShopgateCategoryExtensionPipeline}
    */
   static create (context) {
+    const shopifyAdminFactory = new ShopifyAdminClientFactory(context.config.shopifyAccessToken, context.config.shopifyShopAlias)
```
Comment (Menes1337) [inline extension/lib/shopgate/CategoryExtensionPipeline.js]: Isn't context.config.shopifyShopAlias just context.config.alias?

File: extension/lib/shopify/AdminClient.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+   * @param {string} adminAccessToken
+   * @param {string} shopName
+   */
+  constructor (adminAccessToken, shopName) {
+    super({shopName: shopName, accessToken: adminAccessToken})
+  }
+
+  /**
+   * @param {string} id
+   * @return {Promise.<void>}
```
Comment (Menes1337) [inline extension/lib/shopify/AdminClient.js]: please replace all "Promise.<" with "Promise<"

File: extension/lib/shopify/AdminClient.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+   */
+  async getProductCountByCollectionId (id) {
+    return this.product.count({collection_id: id})
+  }
+
+  /**
+   * @return {Promise.<*>}
```
Comment (Menes1337) [inline extension/lib/shopify/AdminClient.js]: can be Promise<string>, right?

File: extension/lib/shopify/AdminClient.js [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+   */
+  async createStorefrontAccessToken () {
+    return this.storefrontAccessToken.create({title: STOREFRONT_ACCESS_TOKEN_TITLE})
+  }
+
+  /**
+   * @return {Promise.<void>}
```
Comment (Menes1337) [inline extension/lib/shopify/AdminClient.js]: Can be Promise<string[]>?

File: extension/lib/shopify/AdminClient.js (hunk 2 above)
Comment (Menes1337) [reply extension/lib/shopify/AdminClient.js]: It can also be changed to "number" instead of void

**Review by Menes1337 — CHANGES_REQUESTED** [review]
I added a few comments

File: extension/lib/shopify/AdminClient.js (hunk 2 above)
Comment (aaron-martin) [reply extension/lib/shopify/AdminClient.js]: done

File: extension/lib/shopify/AdminClient.js (hunk 3 above)
Comment (aaron-martin) [reply extension/lib/shopify/AdminClient.js]: done

File: extension/lib/shopify/AdminClient.js (hunk 4 above)
Comment (aaron-martin) [reply extension/lib/shopify/AdminClient.js]: done

**Review by Menes1337 — APPROVED** [review]

**Pull request merged**